packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -284,12 +286,16 @@ export class TreeActions {
   }
 }
 
-function collectTreePartsUntil(untilNotifier: Observable<void>, rootNode?: TreeModelRootNode) {
+function collectTreePartsUntil(untilNotifier: Observable<void>, parent: TreeModelHierarchyNode | TreeModelRootNode, rootNode?: TreeModelRootNode) {
   return (source: Observable<LoadedTreePart>) =>
     source.pipe(
       reduce<LoadedTreePart, TreeModel>(
         (treeModel, loadedPart) => {
-          addNodesToModel(treeModel, loadedPart);
+          if (!isTreeModelHierarchyNode(loadedPart.loadedNodes[0])) {
+            addErrorToModel(treeModel, loadedPart.loadedNodes[0], parent);
+          } else {
```

Comment:
as mentioned in the `TreeLoader.ts` comment, I don't like this part - I think we should throw on error instead of returning them in the nodes' array

Reply:
Small adjustments were made for this part to meet new loader output format.